src/sage/schemes/curves/plane_curve_arrangement.py

```diff
@@ -0,0 +1,1371 @@
+        """
+        return self._curves
+
+    def _repr_(self):
+        r"""
+        String representation for a curve arrangement.
+
+        OUTPUT:
+
+        A string.
+
+        EXAMPLES::
+
+            sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: h = H([x * y, x + y + 1, x^3 - y^5, x^2 * y^2 + x^5 + y^5, (x^2 + y^2)^3 + (x^3 + y^3 - 1)^2])
+            sage: h
+            Arrangement of 5 curves in Affine Space of dimension 2 over Rational Field
+            sage: H(())
+            Empty curve arrangement in Affine Space of dimension 2 over Rational Field
+            sage: H.<x, y, z> = ProjectivePlaneCurveArrangements(QQ)
+            sage: h = H([x * y, x + y + z, x^3 * z^2 - y^5, x^2 * y^2 * z + x^5 + y^5, (x^2 + y^2)^3 + (x^3 + y^3 - z^3)^2])
+            sage: h
+            Arrangement of 5 curves in Projective Space of dimension 2 over Rational Field
+
+        """
+        if len(self) == 0:
+            return 'Empty curve arrangement in {0}'.format(self.parent().ambient_space())
+        elif len(self) < 5:
+            curves = ', '.join(h.defining_polynomial()._repr_()
+                               for h in self._curves)
+            return 'Arrangement ({0}) in {1}'.format(curves,
+                                                     self.parent().ambient_space())
+        return 'Arrangement of {0} curves in {1}'.format(len(self),
+                                                         self.parent().ambient_space())
+
+    def _richcmp_(self, other, op):
+        """
+        Compare two curve arrangements.
+
+        EXAMPLES::
+
+            sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: H(x) == H(y)
+            False
+            sage: H(x) == H(2 * x)
+            True
+
+        TESTS::
+
+            sage: H(x) == 0
+            False
+        """
+        return richcmp(self._curves, other._curves, op)
+
+    def union(self, other):
+        r"""
+        The union of ``self`` with ``other``.
+
+        INPUT:
+
+        - ``other`` -- a curve arrangement or something that can
+          be converted into a curve arrangement
+
+        OUTPUT:
+
+        A new curve arrangement.
+
+        EXAMPLES::
+
+            sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: h = H([x * y, x + y + 1, x^3 - y^5, x^2 * y^2 + x^5 + y^5, (x^2 + y^2)^3 + (x^3 + y^3 - 1)^2])
+            sage: C = Curve(x^8 - y^8 -x^4 * y^4)
+            sage: h1 = h.union(C); h1
+            Arrangement of 6 curves in Affine Space of dimension 2 over Rational Field
+            sage: h1 == h1.union(C)
+            Repeated curve
+            True
+        """
+        P = self.parent()
+        other_h = P(other)
+        curves0 = self._curves + other_h._curves
+        curves = ()
+        for h in curves0:
+            if h not in curves:
+                curves += (h, )
+            else:
+                print("Repeated curve")
+        result = P(*curves)
```

Comment:
```suggestion
        curves = []
        for h in curves0:
            if h not in curves:
                curves.append(h)
            else:
                print("Repeated curve")
        result = P(*curves)
```
The print statement should probably be done with a verbosity check in some form (whether the global one or a local `verbose=True` argument). Although I don't really see a point in having the print statement.

Reply:
No more print